packages/php-wasm/universal/src/lib/fs-helpers.ts

```diff
@@ -198,6 +198,45 @@ export class FSHelpers {
 		return FS.isFile(FS.lookupPath(path).node.mode);
 	}
 
+	/**
+	 * Creates a symlink in the PHP filesystem.
+	 *
+	 * @param FS
+	 * @param target
+	 * @param link
+	 */
+	static symlink(FS: Emscripten.RootFS, target: string, link: string): any {
+		return FS.symlink(target, link);
+	}
+
+	/**
+	 * Checks if a path is a symlink in the PHP filesystem.
+	 *
+	 * @param FS
+	 * @param path
+	 * @returns True if the path is a symlink, false otherwise.
+	 */
+	static isSymlink(FS: Emscripten.RootFS, path: string): boolean {
+		if (!FSHelpers.fileExists(FS, path)) {
+			return false;
+		}
+
+		return FS.isLink(FS.lookupPath(path).node.mode);
+	}
+
+	/**
+	 * Reads the target of a symlink in the PHP filesystem.
+	 * @param FS
+	 * @param path
+	 * @returns The target of the symlink.
+	 */
+	static readlink(FS: Emscripten.RootFS, path: string): string {
+		if (!FSHelpers.isSymlink(FS, path)) {
+			throw new Error(`"${path}" is not a symlink`);
```

Comment:
The [PHP realpath()](https://www.php.net/manual/en/function.realpath.php) and [POSIX realpath()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/realpath.html) do not panic if a symlink cannot be resolved. PHP returns `false`, and POSIX returns `NULL` and sets errno.

It would be good to follow that pattern here.

Reply:
For consistency, we can remove the check and just return whatever NODEFS readlink would return here. In case of an error, that would still be an exception ( of type `ErrnoError` ) but it is consistent with all the other methods in the helper file.

Reply:
Ah, I see what you mean. Yeah, I think we probably want to remain consistent with the rest of the FSHelpers.